packages/now-cli/test/dev-server.unit.js

```diff
@@ -175,6 +175,26 @@ test(
     t.is(parsed.query['route-param'], 'b');
   })
 );
+test(
+  '[DevServer] Maintains query when builder defines routes through rewrites',
+  testFixture('now-dev-next-rewrites', async (t, server) => {
+    const res = await fetch(`${server.address}/something?url-param=a`);
+    validateResponseHeaders(t, res);
+
+    const text = await res.text();
+
+    // Hacky way of getting the page payload from the response
+    // HTML since we don't have a HTML parser handy.
```

Comment:
Why not use `pages/api` instead to return a proper json response? Or is this bug only affecting non-api endpoints?

Reply:
Sure, as far as I'm aware the test affects any rewritten  route that targets next, and the failure occurs way before reaching next :) as for the rest of the comments you made on the test, it was cloned from another test which did that and I didn't need to change that specific part :)